src/Switch-MBRAUN_SCU101/main.py

```diff
@@ -0,0 +1,220 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# SweepMe! device class
+# Type: Switch
+# Device: MBRAUN SCU101
+
+from __future__ import annotations
+
+import minimalmodbus
+from EmptyDeviceClass import EmptyDevice  # Class comes with SweepMe!
+
+
+class Device(EmptyDevice):
+    ## here you can add html formatted description to your device class that is shown by modules like 'Logger' or 'Switch' that have a description field.
+    description = """
+        <h3>MBRAUN SCU 101 Shutter</h3>
+        <p>This driver controls the MBRAUN Shutter Control Unit SCU 101.</p>
+        <h4>Hardware Setup</h4>
+        <p>In the SCU 101 menu:</p>
+        <ul>
+        <li>E0: Set the MB Address</li>
+        <li>E1-E2: Check baudrate (19200) and parity (E)</li>
+        <li>E4: Set MB Type to RS 232</li>
+        </ul>
+        <h4>Parameters</h4>
+        <ul>
+        <li>Set the MB Address according to your hardware.</li>
+        <li>Choose the shutter number you want to control (1 or 2)</li>
+        <li>The SweepValue that sets the shutter state can be either int (1 = open,&nbsp;&ne;1 = close), boolean (True = open, False = close), or string ("open", &ne;"open" = close).</li>
+        </ul>
+    """
+
+    def __init__(self) -> None:
+        EmptyDevice.__init__(self)
+
+        self.shortname = "SCU101"  # short name will be shown in the sequencer
+        self.variables = ["State"]  # define as many variables you need
+        self.units = [""]  # make sure that you have as many units as you have variables
+        self.plottype = [True]  # True to plot data, corresponding to self.variables
+        self.savetype = [True]  # True to save data, corresponding to self.variables
+
+        self.port_types = ["COM"]
+
+        # Device Parameter
+        self.port_string: str
+        self.address: int
+        self.modbus: None
+        self.instance_key = None
+
+        # Device States
+        self.shutter_number: int
+        self.shutter_state: int
+        self.shutter_state_dict = {
+            0: "Unknown",
+            1: "Open",
+            2: "Closed",
+            3: "Moving",
+            4: "Not connected",
+            5: "Blocked",
+        }
+
+    def set_GUIparameter(self) -> dict:
+        """Get parameters from the GUI and set them as attributes."""
+        return {
+            "SweepMode": ["State", "None"],
+            "MB Address": "1",
+            "Shutter Number": ["1", "2"],
```

Comment:
The Switch module also has the hidden "Channel" field that is shown when defining the key "Channel" here. The Channel could be used instead of the shutter number, because it is typically used to set a sub-unit of an instrument.
Of course, in case of a Switch, creating an own parameter also works.